src/plugins/intel_gpu/include/intel_gpu/primitives/primitive.hpp

```diff
@@ -297,8 +300,46 @@ struct primitive {
         }
     }
 
+    input_info& get_dependency(size_t idx) {
+        if (idx < input.size())
+            return input[idx];
+
+        auto dependencies_map = get_dependencies_map();
+        OPENVINO_ASSERT(dependencies_map.count(idx) > 0,
+                        "[GPU] Requested index ",
+                        std::to_string(idx),
+                        " exceeds total dependencies count (",
+                        std::to_string(input.size() + dependencies_map.size()),
+                        ") for",
+                        id,
+                        " primitive");
+
+        // get_dependencies_map() returns `const input_info*` for general read-only access.
+        // However since the current function is non-const and the object itself is not actually const,
+        // we can safely cast away constness to return a mutable reference.
+        // This avoids duplicating the dependencies logic while preserving const correctness.
```

Comment:
actually, this is not true, because some `input_info` are const.(for example, convolution) What about just removing `const` keyword from such primitives as well?

Reply:
Removed const for Convolution's input_infos